src/aiida_quantumespresso/parsers/parse_raw/pw.py

```diff
@@ -921,6 +923,23 @@ def parse_stdout(stdout, input_parameters, parser_options=None, parsed_xml=None)
     parsed_data['structure'] = structure_data
     parsed_data['trajectory'] = trajectory_data
 
+    # Double check to detect error messages if CRASH is found
+    if crash_file is not None:
+        lines = crash_file.split('\n')
+
+        for line_number, line in enumerate(lines):
+            # Compare the line to the known set of error and warning messages and add them to the log container
+            detect_important_message(logs, line)
+
+        if len(logs.error) or len(logs.warning) > 0:
+            parsed_data['bands'] = bands_data
+            parsed_data['structure'] = structure_data
+            parsed_data['trajectory'] = trajectory_data
+            return parsed_data, logs
+
+        # did not find any error message -> raise an Error and do not return anything
+        raise QEOutputParsingError('Parser cannot load basic info.')
```

Comment:
With the other comments in mind, I would change this maybe to the following. Note that I added the lines to deduplicate the log messages. These should be removed from lines 919 - 920
```suggestion

    # Double check to detect error messages if CRASH is found
    if crash_file is not None:
        for line_number, line in enumerate(crash_file.split('\n')):
            # Compare the line to the known set of error and warning messages and add them to the log container
            detect_important_message(logs, line)

        if len(logs.error) == 0 and len(logs.warning) == 0:
            raise QEOutputParsingError('The calculation crashed but the relevant error could not be parsed.')

    # Remove duplicate log messages by turning it into a set. Then convert back to list as that is what is expected
    logs.error = list(set(logs.error))
    logs.warning = list(set(logs.warning))
```